.gitignore

```diff
@@ -2,7 +2,6 @@
 *.egg/
 *.pyc
 *.pyo
-*.cpp
```

Comment:
Ideep is a header-only library. I'm wondering what is the story of building if we add cpp source files to it.

Reply:
The current building system in pytorch will only use IDEEP_INCLUDE_DIR (ideep/include/), and MKLDNN_INCLUDE_DIR (ideep/mkl-dnn), if I understand correctly. So the newly added ideep/python/ dir will not break the current building system. But apparently additional work is needed to enable python binding in pytorch.

Reply:
Alternatively, if we'd like to enforce the existing `.gitignore` policy, then we can merge the `hpp` & `cpp` files into the `hpp` header, since it's only being included in one file in the bridge. Thanks!

Reply:
How do we plan to integrate this into PyTorch? Currently, ideep is header file only and the build target is defined by oneDNN. What target would this newly added source file be put in, libtorch_cpu.so? Would it be a weird design that PyTorch links a source file, not a static/dynamic library, from a third-party?

Reply:
Good point, thanks! The design is indeed weird.
We can merge the `cpp` & `hpp` file in this PR into one header file for these bindings, since they would only be included in one file.

BTW, as a digression, could you please suggest if there's a way to make the integration work if the bindings would be in `torch_cpu` instead of `torch_python`? Thanks!

Reply:
> BTW, as a digression, could you please suggest if there's a way to make the integration work if the bindings would be in `torch_cpu` instead of `torch_python`? I had intended to use them in the target `torch_python`, but doing so will cause `torch_cpu` & `torch_python` to be linked to oneDNN Graph separately, causing a regression with LLGA JIT tracer (not with Inductor LLGA bridge, though). Thanks!

From PyTorch, the python binding should go with `torch_python` actually... I don't think linking it to `torch_cpu` is the right thing to do. My apology for the confusing message previously saying it does to `libtorch_cpu.so`.

Reply:
Thanks for confirming, Jiong!

@ElaineBao, can we proceed with having just a header file in this PR? Thanks!

Reply:
Thanks for the great discussion! I've updated the PR to only include one header file. Also add co-authors info according to Sanchit's suggestion :)